aptos-move/block-executor/src/modules_and_scripts.rs

```diff
@@ -0,0 +1,300 @@
+    }
+
+    fn fetch_module_size_in_bytes(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+    ) -> VMResult<Option<usize>> {
+        Ok(self
+            .get_module_storage_entry(address, module_name)?
+            .into_module_module_storage_entry_at_idx()
+            .map(|(_, entry)| entry.bytes().len()))
+    }
+
+    fn fetch_module_metadata(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+    ) -> VMResult<Vec<Metadata>> {
+        Ok(self
+            .get_existing_module_storage_entry_with_idx(address, module_name)?
+            .1
+            .metadata()
+            .to_vec())
+    }
+
+    fn fetch_deserialized_module(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+    ) -> VMResult<Arc<CompiledModule>> {
+        Ok(self
+            .get_existing_module_storage_entry_with_idx(address, module_name)?
+            .1
+            .as_compiled_module())
+    }
+
+    fn fetch_verified_module(
+        &self,
+        address: &AccountAddress,
+        module_name: &IdentStr,
+    ) -> VMResult<Arc<Module>> {
+        let mut visited = HashSet::new();
+        self.traversed_published_dependencies(address, module_name, &mut visited)
+    }
+}
+
+impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<'a, T, S, X> {
+    /// Returns the module storage entry built from the base (storage) view. If there is a
+    /// storage error, returns an error.
+    fn get_base_module_storage_entry(
+        &self,
+        key: &T::Key,
+    ) -> VMResult<Option<Arc<ModuleStorageEntry>>> {
+        self.get_raw_base_value(key)
+            .map_err(|e| e.finish(Location::Undefined))?
+            .map(|s| {
+                ModuleStorageEntry::from_state_value(self.runtime_environment, s).map(Arc::new)
+            })
+            .transpose()
+    }
+
+    /// Returns the module storage entry built from the current view. If it is not in
+    /// multi-version or non-sync data structures, fetches it from the base view.
+    fn get_module_storage_entry_by_key(&self, key: &T::Key) -> VMResult<ModuleStorageReadResult> {
+        use ModuleStorageReadResult::*;
+
+        match &self.latest_view {
+            ViewState::Sync(state) => {
+                let result = state.versioned_map.module_storage().get_or_else(
+                    key,
+                    ShiftedTxnIndex::new(self.txn_idx),
+                    || self.get_base_module_storage_entry(key),
+                )?;
+                state
+                    .captured_reads
+                    .borrow_mut()
+                    .capture_module_storage_read(key.clone(), &result);
+                Ok(result)
+            },
+            ViewState::Unsync(state) => {
+                state.read_set.borrow_mut().module_reads.insert(key.clone());
+                Ok(match state.unsync_map.fetch_module(key) {
```

Comment:
hm, it this flipped? shouldn't here be v2 and in view.rs the normal fetch_module?

Reply:
I flipped implementations, this uses the new flow, old API is renamed to `if_using_...`